src/librbd/internal.cc

```diff
@@ -3826,4 +3835,212 @@ int mirror_image_disable_internal(ImageCtx *ictx, bool force,
 
     return 0;
   }
+
+  int group_image_add(librados::IoCtx& group_ioctx, const char *group_name,
+		      librados::IoCtx& image_ioctx, const char *image_name)
+  {
+    CephContext *cct = (CephContext *)group_ioctx.cct();
+    ldout(cct, 20) << "group_image_add " << &group_ioctx
+                   << " group name " << group_name << " image "
+        	   << &image_ioctx << " name " << image_name << dendl;
+
+    string group_id;
+
+    int r = cls_client::dir_get_id(&group_ioctx, RBD_GROUP_DIRECTORY, group_name, &group_id);
+    if (r < 0) {
+      lderr(cct) << "error reading consistency group id object: "
+                 << cpp_strerror(r)
+        	 << dendl;
+      return r;
+    }
+    string group_header_oid = util::group_header_name(group_id);
+
+    ldout(cct, 20) << "adding image to group name " << group_name
+                   << " group id " << group_header_oid << dendl;
+
+    ImageCtx *imctx = new ImageCtx(image_name, "", "", image_ioctx, true);
```

Comment:
I need header_oid below for referencing the group. Can I get it without opening the image?